receiver/mongodbatlasreceiver/internal/mongodb_atlas_client.go

```diff
@@ -0,0 +1,104 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+
+	"github.com/mongodb-forks/digest"
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.uber.org/zap"
+)
+
+// This struct wraps the official MongoDB Atlas client to manage pagination
+// and mapping to OpenTelmetry metric and log structures.
+type MongoDBAtlasClient struct {
+	log    *zap.Logger
+	client *mongodbatlas.Client
+}
+
+func NewMongoDBAtlasClient(
+	publicKey string,
+	privateKey string,
+	log *zap.Logger,
+) (*MongoDBAtlasClient, error) {
+	t := digest.NewTransport(publicKey, privateKey)
+	tc, err := t.Client()
+	if err != nil {
+		return nil, err
```

Comment:
Wrapping with some context can be helpful where logging isn't performed.  There are currently a couple similar scenarios where raw errors are returned.

```suggestion
		return nil, fmt.Errorf("could not create Mongo transport http client: %w", err)
```

Reply:
Thanks for this. I think I have all the current errors wrapped in the right places now.